src/metabase/sync/analyze/fingerprint/insights.clj

```diff
@@ -95,25 +95,28 @@
 
 (def ^:private ^:const ^Long validation-set-size 20)
 
+(defn- numeric?
+  [x]
+  (not (or (nil? x)
+           (Double/isNaN x)
+           (Double/isInfinite x))))
```

Comment:
This will throw an Exception if you pass something that's not numeric, which is not they way you'd normally expect a predicate function to work. We also have `clojure.core/number?` already, but since this is testing something different you should give it a clearer name. (It's difficult to guess that `numeric?` is tests that something is a number, but not infinity/NaN, without looking at the source.) I would change it to something like this:

```clj
(defn- real-number?
  [x]
  (and (number? x)
       (not (Double/isNaN x))
       (not (Double/isInfinite x))))
```